### PR TITLE
Convert next-server.js to typescript

### PR DESCRIPTION
**Pull request by timneutkens**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   "standard": {
     "parser": "babel-eslint",
     "ignore": [
+      "packages/next-server/server/next-server.ts",
```
Comment (timneutkens) [inline package.json]: The standardjs/eslint parser for typescript couldn't parse the file. So it's time to implement `tslint`. I'll create a good first issue for it after this PR is merged

**Review by rauchg — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (shmidt-i) [reply package.json]: You can create a separate override section with typescript-eslint-parser for .ts, .tsx files

Comment (timneutkens) [reply package.json]: It's already using the ts parser, it's just not parsing public/private correctly

File: packages/next-server/server/next-server.ts [hunk 2]
```diff
@@ -78,44 +103,44 @@ export default class Server {
       })
   }
 
-  getRequestHandler () {
+  public getRequestHandler () {
```
Comment (steida) [inline packages/next-server/server/next-server.ts]: We don't have to write public, it's a default. 
https://github.com/este/este-typescript/blob/master/tslint.json#L11

**Pull request merged**